agent/util-scripts/pbench-tool-meister

```diff
@@ -0,0 +1,836 @@
+            tar_dir = Path(self._tmp_dir, self._hostname)
+            try:
+                # Invoke tar directly for efficiency.
+                retcode = os.spawnl(
+                    os.P_WAIT, "/usr/bin/tar", "tar", "-Jcf", tar_file, tar_dir,
+                )
+            except Exception:
+                self.logger.exception("Failed to create tools tar ball")
+            else:
+                if retcode != 0:
+                    self.logger.error(
+                        "Failed to create tools tar ball;" " return code: %d", retcode,
+                    )
+                else:
+                    try:
+                        tar_md5 = md5sum(tar_file)
+                    except Exception:
+                        self.logger.exception("Failed to read tools tar ball")
+                    else:
+                        self.logger.info(
+                            "%s: send_tool %s %s",
+                            self._hostname,
+                            self._group,
+                            self._directory,
+                        )
+                        headers = {"http-md5sum": tar_md5}
+                        directory_bytes = data["directory"].encode("utf-8")
+                        tool_data_ctx = hashlib.md5(directory_bytes).hexdigest()
+                        url = f"http://{self._controller}:8080/tool-data/{tool_data_ctx}/{self._hostname}"
+                        with tar_file.open("rb") as tar_fp:
+                            response = requests.put(url, headers=headers, data=tar_fp)
+                            if response.status_code != 200:
+                                self.logger.error(
+                                    "PUT '%s' failed with '%d', '%s'",
+                                    url,
+                                    response.status_code,
+                                    response.text,
+                                )
+                            else:
+                                self.logger.debug(
+                                    "PUT '%s' succeeded ('%d', '%s')",
+                                    url,
+                                    response.status_code,
+                                    response.text,
+                                )
+                                try:
+                                    shutil.rmtree(tar_dir)
+                                except Exception:
+                                    self.logger.exception(
+                                        "Failed to remove tool data hierarchy, '%s'",
+                                        tar_dir,
+                                    )
+            finally:
+                try:
+                    tar_file.unlink()
+                except Exception:
+                    self.logger.warning(
+                        "failed to remove tools tar ball, '%s'", tar_file
+                    )
+        self._directory = None
+        return failures
+
+
+def main(argv):
```

Comment:
This part should be in /lib/pbench/cli

Reply:
Agreed.  This work started before that infrastructure was created, and would like to avoid more changes since it is already 280+ files.

This kind of restructuring can be done in a follow-on commit.